src/parse-markdown/index.ts

```diff
@@ -9,6 +9,14 @@ export default function parseMarkdown(md: string, fname: string) {
   const hasTitle =
     paragraphs[0].length > 0 && paragraphs[1] === "" && paragraphs[2] === "";
 
+  // Check for alternative Heading Two syntax
+  for (let i = 0; i < paragraphs.length; i++) {
```

Comment:
Can eliminate checking for condition `i < paragraphs.length - 1` by limiting the iteration condition.

```ts
for (let i = 0; i < (paragraphs.length - 1); i++) {
  if (i < paragraphs[i + 1].match(/^( {0,3}-+\s*)$/) !== null) {
    paragraphs[i] = `${paragraphs[i]}\n${paragraphs[i + 1]}`;
    paragraphs[i + 1] = "";
  }
}
```